diff --git a/node_modules/ghost/core/frontend/helpers/t.js b/node_modules/ghost/core/frontend/helpers/t.js
index 8438481..765a636 100644
--- a/node_modules/ghost/core/frontend/helpers/t.js
+++ b/node_modules/ghost/core/frontend/helpers/t.js
@@ -12,14 +12,18 @@
 
 const {themeI18n} = require('../services/handlebars');
 
-module.exports = function t(text, options) {
-    const bindings = {};
-    let prop;
-    for (prop in options.hash) {
-        if (Object.prototype.hasOwnProperty.call(options.hash, prop)) {
-            bindings[prop] = options.hash[prop];
-        }
+module.exports = function t(textObject, options) {
+  const text = typeof textObject === "string" ? textObject : textObject.string;
+  const lang = options.data.root.blockCache?.lang?.[0];
+  const path = lang ? `${lang}.${text}` : text;
+
+  const bindings = {};
+  let prop;
+  for (prop in options.hash) {
+    if (Object.prototype.hasOwnProperty.call(options.hash, prop)) {
+      bindings[prop] = options.hash[prop];
     }
+  }
 
-    return themeI18n.t(text, bindings);
+  return themeI18n.t(path, bindings);
 };
